src/structures/ClientApplication.js

```diff
@@ -5,41 +5,82 @@ const Application = require('./interfaces/Application');
 
 /**
  * Represents a Client OAuth2 Application.
+ * <info>The only guaranteed properties are `id` and `flags`. To get the entire
+ * application, you need to fetch it with `application.fetch()` first.</info>
  * @extends {Application}
  */
 class ClientApplication extends Application {
   _patch(data) {
     super._patch(data);
 
-    /**
-     * The app's cover image
-     * @type {?string}
-     */
-    this.cover = data.cover_image || null;
-
-    /**
-     * The app's RPC origins, if enabled
-     * @type {string[]}
-     */
-    this.rpcOrigins = data.rpc_origins || [];
-
-    /**
-     * If this app's bot requires a code grant when using the OAuth2 flow
-     * @type {?boolean}
-     */
-    this.botRequireCodeGrant = typeof data.bot_require_code_grant !== 'undefined' ? data.bot_require_code_grant : null;
-
-    /**
-     * If this app's bot is public
-     * @type {?boolean}
-     */
-    this.botPublic = typeof data.bot_public !== 'undefined' ? data.bot_public : null;
-
-    /**
-     * The owner of this OAuth application
-     * @type {?User|Team}
-     */
-    this.owner = data.team ? new Team(this.client, data.team) : data.owner ? this.client.users.add(data.owner) : null;
+    if ('flags' in data) {
+      /**
+       * The flags this application has
+       * @type {number}
+       */
+      this.flags = data.flags;
+    }
+
+    if ('cover_image' in data) {
+      /**
+       * The app's cover image
+       * @type {string}
+       */
+      this.cover = data.cover_image;
+    }
+
+    if ('rpc_origins' in data) {
+      /**
+       * The app's RPC origins, if enabled
+       * @type {string[]}
+       */
+      this.rpcOrigins = data.rpc_origins;
+    }
+
+    if ('bot_require_code_grant' in data) {
+      /**
+       * If this app's bot requires a code grant when using the OAuth2 flow
+       * @type {boolean}
+       */
+      this.botRequireCodeGrant = data.bot_require_code_grant;
+    }
+
+    if ('bot_public' in data) {
+      /**
+       * If this app's bot is public
+       * @type {boolean}
+       */
+      this.botPublic = data.bot_public;
+    }
+
+    if ('team' in data) {
+      /**
+       * The owner of this OAuth application
+       * @type {User|Team}
+       */
+      this.owner = new Team(this.client, data.team);
+    } else if ('owner' in data) {
+      this.owner = this.client.users.add(data.owner);
+    }
+  }
```

Comment:
Was nullable before, decided to make them optional instead to be consistent with other partial structures (see https://github.com/discordjs/discord.js/pull/5147#discussion_r548765022)
I decided to add a notice at the top of the `ClientApplication` class instead of adding it to the jsdoc type so that it doesn't show undefined for integration applications, but I can change that if it's preferred